### PR TITLE
Fixed the Riot foam icon

### PR DESCRIPTION
**Pull request by shazbot194**
Adds the variable riot to both normal foam darts and adds two else if statements and modifies one if statement to give both darts and both modified darts the correct icon.

### CODE REVIEW

**Comment by Fox-McCloud** [comment]
This is all...super unnecessary....can be simplified to something much better and future-proof.

File: code/modules/projectiles/ammunition/ammo_casings.dm [hunk 1]
```diff
@@ -306,19 +306,30 @@
...
 		desc = "Its nerf or nothing! ... Although, this one doesn't look too safe."
 		if(BB)
 			BB.icon_state = "foamdart_empty"
+	else if(riot)
+		icon_state = "foamdart_riot"
+		desc = "Whose smart idea was it to use toys as crowd control? Ages 18 and up."
+		if(BB)
+			BB.icon_state = "foamdart_riot"
 	else
 		icon_state = "foamdart"
```
Comment (Fox-McCloud) [inline code/modules/projectiles/ammunition/ammo_casings.dm]: `icon_state = initial(icon_state)`

File: code/modules/projectiles/ammunition/ammo_casings.dm [hunk 2]
```diff
@@ -306,19 +306,30 @@
...
 		desc = "Its nerf or nothing! Ages 8 and up."
 		if(BB)
-			BB.icon_state = "foamdart_empty"
+			BB.icon_state = "foamdart"
```
Comment (Fox-McCloud) [inline code/modules/projectiles/ammunition/ammo_casings.dm]: `BB.icon_state = initial(BB.icon_state)`

**Comment by shazbot194** [comment]
How about replcing the if, else if, and else block with this;
/obj/item/ammo_casing/caseless/foam_dart/update_icon()
	..()
	if(modified)
		icon_state = "foamdart_empty"
		if(riot)
			desc = "Whose smart idea was it to use toys as crowd control? ... Although, this one doesn't look too safe."
		else
			desc = "Its nerf or nothing! ... Although, this one doesn't look too safe."
		if(BB)
			BB.icon_state = "foamdart_empty"
	else
		icon_state = initial(icon_state)
		if(BB)
			BB.icon_state = initial(BB.icon_state)

It is smaller then the one that is currently on the server, slightly more future proof then that one as well and has no bugs, and uses all the proper sprites.

**Comment by Fox-McCloud** [comment]
If you implemented what I stated, it's completely future proof for other colored darts, as long as the icon state on the foam dart and casing are properly set on the subtype.

**Comment by shazbot194** [comment]
Is there a way to edit this pull or will I have to make a new one to submit the changes?

**Comment by Fox-McCloud** [comment]
Are you...using the git text editor?

**Comment by shazbot194** [comment]
I am using dream maker but I just found the Git editor, I did just test this code however and it does work.

File: code/modules/projectiles/ammunition/ammo_casings.dm [hunk 3]
```diff
@@ -306,19 +306,22 @@
...
 
 /obj/item/ammo_casing/caseless/foam_dart/update_icon()
 	..()
 	if(modified)
 		icon_state = "foamdart_empty"
-		desc = "Its nerf or nothing! ... Although, this one doesn't look too safe."
+		if(riot)
+			desc = "Whose smart idea was it to use toys as crowd control? ... Although, this one doesn't look too safe."
+		else
+			desc = "Its nerf or nothing! ... Although, this one doesn't look too safe."
```
Comment (Fox-McCloud) [inline code/modules/projectiles/ammunition/ammo_casings.dm]: The above reallyyyy isn't necessary.

**Comment by shazbot194** [comment]
I agree, I was trying to keep in how the dart's description says it looks unsafe, but without it does cut down on any additional lines if someone wants to add more darts in, shall I just take it out?

**Comment by Fox-McCloud** [comment]
Yeah, I'd just...take it out to be honest--the major problem here are the icons messing up.

**Comment by shazbot194** [comment]
Ok, I took out the changing description and just left what is needed for the icons.

File: code/modules/projectiles/ammunition/ammo_casings.dm [hunk 4]
```diff
@@ -306,19 +306,18 @@
 	icon = 'icons/obj/guns/toy.dmi'
 	icon_state = "foamdart"
 	var/modified = 0
+	var/riot = 0
```
Comment (Fox-McCloud) [inline code/modules/projectiles/ammunition/ammo_casings.dm]: unused var.

File: code/modules/projectiles/ammunition/ammo_casings.dm [hunk 5]
```diff
@@ -306,19 +306,18 @@
...
 
 /obj/item/ammo_casing/caseless/foam_dart/update_icon()
 	..()
 	if(modified)
 		icon_state = "foamdart_empty"
-		desc = "Its nerf or nothing! ... Although, this one doesn't look too safe."
```
Comment (Fox-McCloud) [inline code/modules/projectiles/ammunition/ammo_casings.dm]: Leave this.

**Comment by shazbot194** [comment]
How about making it something along the lines of "This foam dart doesn't look too safe" so its more generic.

**Comment by shazbot194** [comment]
Well I think its just about as future proof as it can get right now, although I doubt its perfect, any more thoughts? I have also tested it, all the icons are now displayed correctly.

**Comment by Crazylemon64** [comment]
`update ammo_casings.dm` :eyes:

**Comment by shazbot194** [comment]
uhhh, that last bit for the stechin pistol isn't meant to be on this one, oops.

**Comment by shazbot194** [comment]
yup, I wasn't sure how to update the pull via dream seeker, but it looks like I did by accident.

**Comment by Fox-McCloud** [comment]
I've said it once, but it looks like I'm going to to have to say it again. If you can't be bothered to set up git and make changes in DM, directly, then don't bother submitting a PR at all. Github editor submitted PRs will be denied.

**Comment by PJB3005** [comment]
@Fox-McCloud and, how does that matter in any way? Code is code.

**Comment by Fox-McCloud** [comment]
Because it means they didn't bother compiling or testing it.

**Comment by shazbot194** [comment]
I did compile it and test it, I even said I wasn't sure how to until I did it by accident. In addition to the multiple times I said I have tested it along the way, if you would like I could republish it from dream maker, but it would be the same exact tested code.

File: code/modules/projectiles/ammunition/ammo_casings.dm [hunk 6]
```diff
@@ -310,15 +310,14 @@
 /obj/item/ammo_casing/caseless/foam_dart/update_icon()
 	..()
 	if(modified)
+		desc = "This foam dart doesn't look too safe"
 		icon_state = "foamdart_empty"
-		desc = "Its nerf or nothing! ... Although, this one doesn't look too safe."
```
Comment (Fox-McCloud) [inline code/modules/projectiles/ammunition/ammo_casings.dm]: Don't change the description or order unless there's an explicit reason to. In this case, there isn't.

**Comment by shazbot194** [comment]
I still think that a the description "Its nerf or nothing! ... Although, this one doesn't look too safe." doesn't fit too well on modified riot darts and the slightly more generic "This foam dart doesn't look too safe" works well for both uncapped foam force, riot darts, and any future darts, but here is it is with the former description.

**Pull request merged**